lib/vdsm/network/nmstate/bridge_opts.py

```diff
@@ -0,0 +1,136 @@
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+#
+# Refer to the README and COPYING files for full details of the license
+#
+
+from __future__ import absolute_import
+from __future__ import division
+
+from vdsm.network.nmstate.schema import LinuxBridge
+
+
+class OptStringParser:
+    def __init__(self, delim=' ', assign_op='='):
+        self._delim = delim
+        self._assign_op = assign_op
+
+    def parse(self, opts_str: str):
+        if opts_str:
+            opts = opts_str.strip(self._delim).split(self._delim)
+            return dict(opt.split(self._assign_op, 1) for opt in opts)
+        return {}
+
+
+_BRIDGE_OPTS = {
+    'mac-ageing-time': int,
+    'group-forward-mask': int,
+    'hash-max': int,
+    'multicast-snooping': bool,
+    'multicast-router': int,
+    'multicast-last-member-count': int,
+    'multicast-last-member-interval': int,
+    'multicast-membership-interval': int,
+    'multicast-querier': bool,
+    'multicast-querier-interval': int,
+    'multicast-query-use-ifaddr': bool,
+    'multicast-query-interval': int,
+    'multicast-query-response-interval': int,
+    'multicast-startup-query-count': int,
+    'multicast-startup-query-interval': int,
+}
+
+
+_STP_OPTS = {
+    'enabled': bool,
+    'forward-delay': 'sec_to_hundredth_sec',
+    'hello-time': 'sec_to_hundredth_sec',
+    'max-age': 'sec_to_hundredth_sec',
+    'priority': int,
+}
+
+
+class BridgeOptsParser(OptStringParser):
+    def __init__(self, opts_by_type: {}):
+        super(BridgeOptsParser, self).__init__()
+        self._opts_by_type = opts_by_type
+        self._not_found = []
+
+    def parse(self, opts: str) -> {}:
+        """
+        param: opts: vdsm.network.nmstate.bridge_util.NetworkConfig bridge opts
+        """
+        str_dict = super().parse(opts.replace('_', '-')) if opts else {}
+        opts_dict = {}
+        for k, v in str_dict.items():
+            try:
+                opts_dict[k] = self._convert_value(k, v)
+            except KeyError:
+                self._not_found.append(k)
+        return opts_dict
+
+    @property
+    def not_found(self):
+        return self._not_found
+
+    def _convert_value(self, key, value):
+        conversion_func = self._opts_by_type[key]
+        if conversion_func == bool:
+            conversion_func = self._str_to_bool
+        elif conversion_func == 'sec_to_hundredth_sec':
+            conversion_func = self._multiply_by_hundred
+        return conversion_func(value)
+
+    @staticmethod
+    def _str_to_bool(str_value):
```

Comment:
Those methods are static, we could move them outside and then we wouldn't need the if else chain checking which conversion it should do. It would also become more flexible that way.

Reply:
done